### ISSUE TITLE
Local caching of vulnerabilities 

### ISSUE THREAD

**Issue by bufferoverflow** [verdaccio/verdaccio-audit#4]
Node security wg is planning to create a dedicated repo for the known vulnerabilities, see 
https://github.com/nodejs/security-wg/issues/359#issuecomment-410079404

Using that sources instead of redirecting to upstream npm could be a nice option for verdaccio-audit.

**Comment by juanpicado** [comment]
Thanks @bufferoverflow for let me know 👍

**Comment by juanpicado** [comment]
@bufferoverflow do you know how to proceed with this? I'm a bit outdated.

**Comment by thib3113** [comment]
I'm also interrested by this, because I'm using dependency-check ... And it .. spam the audit api ...

As I'm understanding, you need to clone the repo ( or add it in dependency, but package.json version seems not updated ) .

After : 
 - read files in the core folder to search vulnerabilities with the nodejs version ( but maybe not appropriate for audit api ).
 - read files in `/ecosystem/<dependency>/` and get the `vulnerable_versions` in the json, compare it with the received version ( I didn't really know what npm send to audit api ) .

The problem, will maybe the need of `git` ( to install the repo ) .

**Comment by bufferoverflow** [comment]
I guess we have to wait until https://github.com/nodejs/security-advisories is shipped as a npm package or on a web site.

**Comment by thib3113** [comment]
@bufferoverflow not sure a npm package was a good idea .

Maybe better to do : 
 - use github api to check if update each 4 (?) hours
 - if update, update local database ( clone/"download archive" in a local folder )

It's what do dependency-check with NVD database ( if database is some hours old, it update it ) . 

I'm not sure they want to do a website with an api for this, I thin they just want to do a website to search a vulnerability, or a package to see if it contain vulnerabilities and wich version

**Comment by bufferoverflow** [comment]
@thib3113 valid points, worth to do!

**Comment by thib3113** [comment]
@bufferoverflow I've just checking a little this way ( before answering on this thread ) .
And, I think I didn't have the time to do it, but this can be usefull for you : 

 - github api to get last master commit : `https://api.github.com/repos/nodejs/security-advisories/commits/master`
 - url to download the zip : `https://github.com/nodejs/security-advisories/archive/master.zip`

**Comment by DanielRuf** [comment]
Not sure but http://registry.npmjs.com/-/v1/search?text=is:insecure&size=250 seems to contain all advisories.

**Comment by DanielRuf** [comment]
https://github.com/nodejs/security-advisories/archive/master.tar.gz would be probably better so save some bandwidth.

**Comment by thib3113** [comment]
@DanielRuf yes, It's what I said ( about the master.tar.gz ... just with a .zip )
